pkg/virt-launcher/virtwrap/network/dhcpv6/dhcpv6.go

```diff
@@ -0,0 +1,67 @@
+package dhcpv6
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/vishvananda/netlink"
+
+	"github.com/insomniacslk/dhcp/dhcpv6"
+	"github.com/insomniacslk/dhcp/dhcpv6/server6"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/log"
+)
+
+const (
+	infiniteLease = 999 * 24 * time.Hour
+)
+
+type DHCPv6Handler struct {
+	clientIP      net.IP
+	leaseDuration time.Duration
+}
+
+func SingleClientDHCPv6Server(
+	clientIP net.IP,
+	serverIface string,
+	dnsIPs [][]byte,
+	routes *[]netlink.Route,
+	searchDomains []string,
+	mtu uint16,
+	customDHCPOptions *v1.DHCPOptions) error {
+
+	log.Log.Info("Starting SingleClientDHCPv6Server")
+
+	handler := &DHCPv6Handler{
+		clientIP:      clientIP,
+		leaseDuration: infiniteLease,
+	}
+
+	s, err := server6.NewServer("", nil, handler.ServeDHCPv6)
+	if err != nil {
+		return fmt.Errorf("couldn't create dhcpv6 server: %v", err)
+	}
+	go func() {
+		_ = s.Serve()
+	}()
+
+	return nil
+}
+
+func (h *DHCPv6Handler) ServeDHCPv6(conn net.PacketConn, peer net.Addr, m dhcpv6.DHCPv6) {
+	log.Log.V(4).Info("Serving a new request")
+
+	// TODO how can we make sure the request is from the vm? Is filtering requests arrived to the bridge interface enough?
```

Comment:
I hope this can be answered before this gets merged. It will not be nice to leave it here, it hints the work is incomplete. Perhaps opening an issue is enough to track this.

Reply:
The comment was changed to - "TODO if we extend the server to support bridge binding, we need to filter out non-vm requests" in the unit test commit. I will move the change to the correct commit.